File: bot/skills/nya.py
import logging

from telegram import Update
from telegram.error import BadRequest
from telegram.ext import Application, ContextTypes
from handlers import ChatCommandHandler

logger = logging.getLogger(__name__)


def add_nya(app: Application, handlers_group: int):
    logger.info("registering nya handlers")
    app.add_handler(
        ChatCommandHandler(
            "nya",
            nya,
            require_admin=True,
        ),
        group=handlers_group,
    )


async def nya(update: Update, context: ContextTypes.DEFAULT_TYPE):
    text = " ".join(context.args)
    chat_id = update.effective_chat.id

    try:
        await context.bot.delete_message(chat_id, update.effective_message.message_id)
    except BadRequest as err:
        logger.info("can't delete msg: %s", err)

    if text:
        await context.bot.send_message(chat_id, text)

File: internal/skill/nya.go
package skill

import (
	"context"
	"log/slog"
	"strings"

	"github.com/go-telegram/bot"
	"github.com/go-telegram/bot/models"

	appbot "github.com/vldc-hq/vldc-bot/internal/bot"
)

func NyaSkill() appbot.Skill {
	return appbot.Skill{
		Name: "nya",
		Hint: "Simon says wat?",
		Register: func(b *bot.Bot, _ *appbot.Deps) {
			b.RegisterHandler(bot.HandlerTypeMessageText, "/nya", bot.MatchTypePrefix, func(ctx context.Context, b *bot.Bot, update *models.Update) {
				if update.Message == nil {
					return
				}
				msg := update.Message
				if !appbot.IsAdmin(ctx, b, msg.Chat.ID, msg.From.ID) {
					return
				}

				// Delete the command message
				_, err := b.DeleteMessage(ctx, &bot.DeleteMessageParams{
					ChatID:    msg.Chat.ID,
					MessageID: msg.ID,
				})
				if err != nil {
					slog.Info("can't delete msg", "error", err)
				}

				// Send args as text (Python behavior: `/nya some text` sends "some text")
				text := strings.TrimSpace(strings.TrimPrefix(msg.Text, "/nya"))
				if text == "" {
					return
				}
				_, _ = b.SendMessage(ctx, &bot.SendMessageParams{
					ChatID: msg.Chat.ID,
					Text:   text,
				})
			})
		},
	}
}
